pkg/expressions/cel.go

```diff
@@ -0,0 +1,141 @@
+package expressions
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+
+	harness "github.com/kudobuilder/kuttl/pkg/apis/testharness/v1beta1"
+)
+
+func buildProgram(expr string, env *cel.Env) (cel.Program, error) {
+	ast, issues := env.Compile(expr)
+	if issues != nil && issues.Err() != nil {
+		return nil, fmt.Errorf("type-check error: %s", issues.Err())
+	}
+
+	prg, err := env.Program(ast)
+	if err != nil {
+		return nil, fmt.Errorf("program construction error: %w", err)
+	}
+
+	return prg, nil
+}
+
+func buildEnv(resourceRefs []harness.TestResourceRef) (*cel.Env, error) {
+	var errs []error
+	for _, resourceRef := range resourceRefs {
+		if err := resourceRef.Validate(); err != nil {
+			errs = append(errs, fmt.Errorf("validation failed for reference '%v': %w", resourceRef.String(), err))
+		}
+	}
+
+	if len(errs) > 0 {
+		return nil, fmt.Errorf("failed to load resource reference(s): %w", errors.Join(errs...))
+	}
+
+	env, err := cel.NewEnv()
```

Comment:
Are you extending the env somewhere ? Looks like you will want to register a bunch of CEL libs to enable features.

Reply:
Any suggestions on what libs are a must-have @eddycharly ?